uwblueprint/feeding-canadian-kids
# Approve onboarding requests

## Zenhub ticket link

[API: Approve Onboarding Request#3
](https://app.zenhub.com/workspaces/fck-627480c10d8974001872b0ef/issues/gh/uwblueprint/feeding-canadian-kids/3)



## Implementation description
Implemented approving onboarding requests and also reset password to the user when they get approved. 



## Steps to test
1. Go to localhost:5000/graphql
2. Create a onboard request. Replace the contactEmail field with your email. Make sure this email is in the firebase or it wont be able to find this email when you try to approve this request.
```mutation testCreateOnboardingRequest {
                        createOnboardingRequest(
                            userInfo:
                                {contactName: "Jane Doe",
                                contactEmail: "YOUR EMAIL",
                                contactPhone: "12345",
                                role: "ASP"
                                }
                        ) {
                            onboardingRequest {
                            id
                            info {
                                contactName,
                                contactEmail,
                                contactPhone,
                                role
                            }
                            dateSubmitted
                            status
                            }
                        }
                    }
```
3. Click run
4. Copy the "id" field from the response
5. Now to approve. Paste the following and replace the id parameter when the one you copied before.
```mutation testApproveOnboardingRequest {
  approveOnboardingRequest(id: "PUT YOUR ID HERE") {
    onboardingRequest {
      id
      status
      info {
        contactName,
        contactEmail,
        contactPhone, 
      }
      userUid
    }
  }
}
```
6. Click run
7. You should get an email to reset your password.



## What should reviewers focus on?
*


## Checklist
